### PR TITLE
feat: skip filesystem for state and ephemeral partitions in the installer

### PR DESCRIPTION
**Pull request by Unix4ever**
Filesystem creation step is moved on the later stage: when Talos mounts
the partition for the first time.
Now it checks if the partition doesn't have any filesystem and formats
it right before mounting.

Additionally refactored mount options a bit:
- replaced separate options with a set of binary flags.
- implemented pre-mount and post-unmount hooks.

And fixed typos in couple of places and increased timeout for `apid ready`.

Signed-off-by: Artem Chernyshev <artem.0xD2@gmail.com>

### CODE REVIEW

**Review by smira — APPROVED** [review]

File: pkg/cluster/check/default.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func DefaultClusterChecks() []ClusterCheck {
 		func(cluster ClusterInfo) conditions.Condition {
 			return conditions.PollingCondition("apid to be ready", func(ctx context.Context) error {
 				return ApidReadyAssertion(ctx, cluster)
-			}, 2*time.Minute, 5*time.Second)
+			}, 5*time.Minute, 5*time.Second)
```
Comment (smira) [inline pkg/cluster/check/default.go:44]: this is much needed fix!

tl;dr is that there's no wait for `bootkube` anymore, so we need to wait more for worker node kubelets to join and get certs

**Comment by smira** [comment]
Feels like e2e-diskimage failed for real

**Comment by Unix4ever** [comment]
> Feels like e2e-diskimage failed for real

yeah, but I've fixed that yesterday. Unfortunately integration tests got stuck in drone though.

**Review by smira — APPROVED** [review]
